File: anavi-door-phat/buzzer/c/beep.c
#include <wiringPi.h>
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>

//Pin 15 on Raspberry Pi corresponds to BCM GPIO 22 and wiringPi pin 3
#define BeepPin 3

void close()
{
	printf("\nTurning off buzzer...\n");
	digitalWrite(BeepPin, LOW);
	exit(0);
}

int main(void)
{
	signal(SIGINT, close);

	if(-1 == wiringPiSetup())
	{
			printf("setup wiringPi failed!");
			return 1;
	}

	//Set GPIO pin
	pinMode(BeepPin, OUTPUT);

	printf("Press CTRL+C to exit.\n");

	//Play a sound until the user closes the app
	while(1)
	{
		digitalWrite(BeepPin, LOW);
		delay(2);
		digitalWrite(BeepPin, HIGH);
		delay(2);
	}

	return 0;
}

File: anavi-door-phat/buzzer/python/beep.py
# coding=utf-8
import signal
import sys
import time

import RPi.GPIO as GPIO

# Piezo buzzer - pin 15 on Raspberry Pi corresponds to BCM GPIO 22
BeepPin = 15


def close(signal, frame):
    GPIO.output(BeepPin, GPIO.LOW)
    sys.exit(0)


# to use Raspberry Pi board pin numbers
GPIO.setmode(GPIO.BOARD)
GPIO.setwarnings(False)

signal.signal(signal.SIGINT, close)

# set up GPIO output channel
GPIO.setup(BeepPin, GPIO.OUT)

print('Press CTRL+C to exit.')
try:
    while True:
        GPIO.output(BeepPin, GPIO.HIGH)
        time.sleep(0.000002)
        GPIO.output(BeepPin, GPIO.LOW)
        time.sleep(0.000002)
finally:
    GPIO.cleanup()
